src/dirac_cwl_proto/job/json_catalog.py

```diff
@@ -0,0 +1,37 @@
+
+from typing import Annotated
+from pydantic import BaseModel, RootModel, BeforeValidator, AnyUrl
+
+
+def validateLFN(value: str) -> str:
+    value = value.removeprefix("LFN:")
+    return value
+
+def validatePFN(value: str) -> str:
+    value = value.removeprefix("PFN:")
+    return value
+
+# Logical File Name such as LFN:/lhcb/MC/2024/HLT2.DST/00327923/0000/00327923_00000533_1.hlt2.dst
+LFN = Annotated[str, BeforeValidator(validateLFN)]
+PFN = Annotated[AnyUrl, BeforeValidator(validatePFN)]
+StorageElementId = str
+ChecksumType = str
+FileChecksum = str
+
+
+class JSONCatalog(RootModel):
+    class CatalogEntry(BaseModel):
+        class Replica(BaseModel):
+            url: PFN
+            se: StorageElementId
+        replicas: list[Replica]
+        size_bytes: int | None = None
+        checksum: dict[ChecksumType, FileChecksum] | None = None
```

Comment:
I'm just wondering if we need to support multiple checksums.
If yes, may be we could validate the `FileChecksum` based on the `ChecksumType`.

Reply:
Well actually this JSON object will be generated programmatically and I don't see any use case where a user would manually create it. Therefore, I am not sure it actually makes much sense to make tests only around the pydantic model.
I think what we want to have and test in this PR is the code generating this JSON object from a CWL input file.

Example:
- I have this input:

```
input-data:
  - class: File
    path: lfn://path/to/result_1.sim
```

And I expect to get the following output from it:

```json
{
  ...
}
```

Any opinion?

Reply:
I'm inclined to support multiple checksum types particularly to support e.g. Dirac/Gaudi File GUIDs, which facilitates converting a replica catalog to pool XML catalog format

I absolutely agree we should validate checksum based on the Checksum type, in which case the checksum field should be a model